hudi-client/src/main/java/org/apache/hudi/metrics/JmxMetricsReporter.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hudi.metrics;
+
+import com.codahale.metrics.MetricRegistry;
+import com.codahale.metrics.jmx.JmxReporter;
+
+import java.io.Closeable;
+
+/**
+ * Used for testing.
+ */
+public class JmxMetricsReporter extends MetricsReporter {
+
+  private final MetricRegistry registry;
+
+  public JmxMetricsReporter(MetricRegistry registry) {
+    this.registry = registry;
+  }
+
+  @Override
+  public void start() {}
+
+  @Override
+  public void report() {}
+
+  @Override
+  public Closeable getReporter() {
+    return JmxReporter.forRegistry(registry).build();
```

Comment:
Could remove `JmxReporter.forRegistry(registry).build();` to construct method to avoid building multi instance?